### ISSUE TITLE
The loss curve of the img-conditioned reconstruction

### ISSUE THREAD

**Issue by EAST-J** [princeton-computational-imaging/Diffusion-SDF#27]
Hi, I modify the code to train the stage1 and stage 2 for single image-conditioned 3D reconstruction. Following the paper's guidance, I employed ResNet18 as the image encoder and utilized a subset of Shapenetv2, encompassing around 8 object categories and 6000 samples. The results of stage 1 reconstruction appear promising; however, during the training of stage 2, it appears that the diffusion loss is challenging to converge. Here are the loss curves. Could you provide insights into this situation? Is it potentially linked to the datasets, or could there be other factors at play?
![1708667704286](https://github.com/princeton-computational-imaging/Diffusion-SDF/assets/51024280/6cbaf44b-513c-4064-ba7e-91d04b455c71)
![image](https://github.com/princeton-computational-imaging/Diffusion-SDF/assets/51024280/058b91fa-4717-4bdb-b40d-34d3038d1510)

**Comment by gene-chou** [comment]
Diffusion losses tend to fluctuate quite a bit. Increase smoothing in the tensorboard plots to get the overall trend. It definitely seems like the losses are still decreasing in your case.

**Comment by EAST-J** [comment]
> Diffusion losses tend to fluctuate quite a bit. Increase smoothing in the tensorboard plots to get the overall trend. It definitely seems like the losses are still decreasing in your case.

Thx for your quickly reply, I plan to extend the training duration to observe the loss more comprehensively. Moreover, I wonder if the diffusion loss reduction related to the dataset size? When I trained on a very small dataset including about 200 samples for text-guided generation, the diff_100 approaches 0 quickly (like code coments: typically diff_100 approaches 0 while diff_1000 can still be relatively high). However, in the aforementioned scenario, the value of diff_100 fluctuates around 0.02 for a long time.

**Comment by gene-chou** [comment]
That's expected. When the dataset is small the model overfits. The larger the dataset the better.

**Comment by EAST-J** [comment]
Thx, I got it. One more little question, In https://github.com/princeton-computational-imaging/Diffusion-SDF/blob/7eb2d6786b8864dc45cd180be8ad5c5b2f8c1f8f/models/diffusion.py#L231-L234, 
`#model_output1 = self.model(model_input, t, pass_cond=0)`
`#model_output2 = self.model(model_input, t, pass_cond=1)`
`#model_output = model_output2*5 - model_output1*4`
`model_output = self.model(model_input, t, pass_cond=1)`
when get the conditional diffusion predictions, you set the guidance strength $w$ to 0. Is this choice informed by experimental results?

**Comment by gene-chou** [comment]
Yes, we just tried a few things and that setup worked fine. This is definitely empirical though, so play around with the hyperparameters see what works!
